wolfcrypt/src/tfm.c

```diff
@@ -38,6 +38,11 @@
 /* in case user set USE_FAST_MATH there */
 #include <wolfssl/wolfcrypt/settings.h>
 
+#ifdef WOLFSSL_TFM_NO_SMALL_STACK
+#undef WOLFSSL_SMALL_STACK
+#undef WOLFSSL_SMALL_STACK_CACHE
```

Comment:
The `WOLFSSL_SMALL_STACK_CACHE` option is not used in `tfm.c`. Can you also run the stack comparison numbers like @douzzer did for #4661 ?

Does the customer really need `WOLFSSL_SMALL_STACK` if they are turning it off for this module?

Also perhaps we need to rethink this and allow for a way to disable specific modules in a larger forward looking way... including changing what was done for #4661?

Example:

```
/* in types.h */
typedef enum WC_MODULE {
    WC_MODULE_TFM = 0x00000001,
    WC_MODULE_ECC = 0x00000002,
} WC_MODULE;

/* in user_settings.h */
#define WOLFSSL_NO_SMALL_STACK (WC_MODULE_ECC | WC_MODULE_TFM)

/* At top of tfm.c */
#if defined(WOLFSSL_NO_SMALL_STACK) && (WOLFSSL_NO_SMALL_STACK & WC_MODULE_TFM)
#undef WOLFSSL_SMALL_STACK
#undef WOLFSSL_SMALL_STACK_CACHE
#endif
```

Reply:
@dgarske that was #4652 that had the stack stats for `WOLFSSL_ECC_NO_SMALL_STACK`.

Reply:
oh and that's quite an intriguing idea you have there, @dgarske -- it never occured to me that one could do that much math at the preprocessor stage.  are we already doing that?  how portable do you think it is?

Reply:
### baseline full smallstack:

`./configure --quiet --disable-jobserver --enable-fips=disabled --enable-all --enable-smallstack --enable-stacksize=verbose`

[base.log](https://github.com/wolfSSL/wolfssl/files/7720502/base.log)

### with `WOLFSSL_TFM_NO_SMALL_STACK`

`./configure --quiet --disable-jobserver --enable-fips=disabled --enable-all --enable-smallstack CFLAGS=-DWOLFSSL_TFM_NO_SMALL_STACK --enable-stacksize=verbose`

[WOLFSSL_TFM_NO_SMALL_STACK.log](https://github.com/wolfSSL/wolfssl/files/7720504/WOLFSSL_TFM_NO_SMALL_STACK.log)

![Screenshot 2021-12-15 090639](https://user-images.githubusercontent.com/37116438/146211380-c2cf60ed-64d0-44ce-ad78-be68f3a26820.jpg)